### ISSUE TITLE
Python 3.11 ExceptionGroup is incorrectly flagged as type value not ExceptionGroup

### ISSUE THREAD

**Issue by tonybaloney** [microsoft/pylance-release#2762]
## Environment data

-   Language Server version: v2022.3.4
-   OS and version: Windows 11
-   Python version: 3.11a7

## Expected behavior

PyLance infers `except*` statements to be of type `ExceptionGroup`

## Actual behavior

PyLance infers exception types as the filter type

## Code Snippet / Additional information

```python
try:
    raise ExceptionGroup("Demo", [ValueError("ooops"), ValueError("Whoopsie")])
except* ValueError as e:
    print(e)
```

![image](https://user-images.githubusercontent.com/1532417/166400379-00495465-0920-43c3-9aec-538796245b66.png)

See https://peps.python.org/pep-0654/ for the reference on the syntax

**Comment by tonybaloney** [comment]
CC @iritkatriel

**Comment by erictraut** [comment]
Thanks for clarifying. When I initially added support for `except*` in pyright, I didn't have access to a runtime that supported it, so I couldn't validate my assumptions.

This will be fixed in the next release of pyright & pylance.

**Comment by debonte** [comment]
@tonybaloney, the fix is available in the latest Pylance insiders build ([2022.5.0-pre.1](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202250-pre1-4-may-2022)). We're not shipping a stable build this week. If you're not using Pylance insiders, you'll see the fix in next week's stable build.

**Comment by tonybaloney** [comment]
Works a charm. Thanks
